src/snowflake/snowpark/session.py

```diff
@@ -1856,11 +1858,23 @@ def sproc(self) -> StoredProcedureRegistration:
         """
         return self._sp_registration
 
+    def _infer_is_return_table(self, sproc_name: str, *args: Any) -> bool:
+        try:
+            arg_types = [convert_sp_to_sf_type(infer_type(arg)) for arg in args]
+            func_signature = f"{sproc_name.upper()}({', '.join(arg_types)})"
+            sproc_desc = self._run_query(f"describe procedure {func_signature}")
+            return_type = sproc_desc[1][1]
+            return return_type.upper().startswith("TABLE")
+        except Exception:
+            pass
+        return False
+
     def call(
         self,
         sproc_name: str,
         *args: Any,
         statement_params: Optional[Dict[str, Any]] = None,
+        _is_return_table: Optional[bool] = None,
```

Comment:
docstring for this arg

Reply:
added

Reply:
Is this an internal parameter not to be used by a user?

Reply:
Yea probably we should make it public

Reply:
When I added this parameter, I did so with the intention of keeping it private. Having this enabled me skip `desc procedure` check when stored procedure is created in session and we know the return type is a table sproc. It also acts as an override to the result from `_infer_is_return_table`. I guess we can make it public if we want users to be able to override the results from `_infer_is_return_table`

Reply:
@sfc-gh-yixie could I get your opinion on this. Should we make `_infer_is_return_table` public?

Reply:
If it's just for performance I'm good not having it as a public API now.
But how about removing it from this function?

```python
def call(
        self,
        sproc_name: str,
        *args: Any,
        statement_params: Optional[Dict[str, Any]] = None,
        ...
):
    self._call(sproc_name, *args, statement_params, ...)

def _call(
        self,
        sproc_name: str,
        *args: Any,
        statement_params: Optional[Dict[str, Any]] = None,
        is_return_table: Optional[bool] = None,
        ...
)
```

The the internal code can use `_call` instead of `call`, and users won't see `_is_return_table `.